#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k_rho", double_t, 0, "Proportional gain for rho", 0.2, 0, 1.0)
gen.add("k_alpha", double_t, 0, "Proportional gain for alpha", 3.6, 0, 5.0)
gen.add("k_beta", double_t, 0, "Proportional gain for beta", 0.1, 0, 1.0)
gen.add("max_lin_vel", double_t, 0, "Maximum allowed linear velocity in m/s", 2.23, 0, 2.23)
gen.add("max_ang_vel", double_t, 0, "Maximum allowed angular velocity in rad/s", 0.89, 0, 0.89)
gen.add("gps_offset", double_t, 0, "GPS offset w.r.t. WAM-V along X-axis", 0.85, 0, 2.0)
gen.add("lidar_offset", double_t, 0, "LIDAR offset w.r.t. WAM-V along X-axis", 0.7, 0, 2.0)
gen.add("caution_distance", double_t, 0, "Caution distance from target pose", 9.0, 0, 10.0)
gen.add("caution_step", double_t, 0, "Step size to decrement caution distance from target pose", 3.0, 0, 10.0)
gen.add("ball_balance", int_t, 0, "Number of balls yet to be shot from ASV", 4, 0, 4)
gen.add("delivery_delay", double_t, 0, "Time delay in seconds from docking to delivery", 7.0, 0, 15.0)

gen.add("debug", bool_t, 0, "Flag to enable/disable debug messages", False)

exit(gen.generate(PACKAGE, "singaboat_vrx", "ScanDockDeliver"))
